### PR TITLE
DDCE-4643 : Add sbt-accessibility-linter

### PR DESCRIPTION
**Pull request by lmugnai**

### CODE REVIEW

File: a11y/uk/gov/hmrc/merchandiseinbaggage/FrontendAccessibilitySpec.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    case newOrExistingView: NewOrExistingView => render(newOrExistingView)
+    case noDeclarationNeededView: NoDeclarationNeededView => render(noDeclarationNeededView)
+    case paymentCalculationView: PaymentCalculationView => render(paymentCalculationView)
+    case previousDeclarationDetailsView: PreviousDeclarationDetailsView => render(previousDeclarationDetailsView)
+    case progressDeletedView: ProgressDeletedView => render(progressDeletedView)
+    case purchaseDetailsExportView: PurchaseDetailsExportView => render(purchaseDetailsExportView)
+    case purchaseDetailsImportView: PurchaseDetailsImportView => render(purchaseDetailsImportView)
+    case removeGoodsView: RemoveGoodsView => render(removeGoodsView)
+    case retrieveDeclarationView: RetrieveDeclarationView => render(retrieveDeclarationView)
+    case reviewGoodsView: ReviewGoodsView => render(reviewGoodsView)
```
Comment (frikit) [inline a11y/uk/gov/hmrc/merchandiseinbaggage/FrontendAccessibilitySpec.scala:96]: also here I see you have not add 3 views:
- case serviceTimeoutView: ServiceTimeoutView                           => render(serviceTimeoutView)
- case sessionExpiredView: SessionExpiredView                           => render(sessionExpiredView)
- case testOnlyDeclarationJourneyPage: TestOnlyDeclarationJourneyPage   => render(testOnlyDeclarationJourneyPage)

In order to make `TestOnlyDeclarationJourneyPage` working you need to add implicit for a form of string like
`private val stringForm: Form[String]             = Form("declarationJourney" -> nonEmptyText)`
`implicit val arbFormString: Arbitrary[Form[String]]                          = fixed(stringForm)`

There is nothing wrong to test these as they dont have errors now

**Review by frikit — CHANGES_REQUESTED** [review]

File: a11y/uk/gov/hmrc/merchandiseinbaggage/FrontendAccessibilitySpec.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import config.AppConfig
+import forms._
+import model.api._
+import uk.gov.hmrc.merchandiseinbaggage.model.api.calculation.CalculationResults
+import uk.gov.hmrc.merchandiseinbaggage.model.core.{PurchaseDetailsInput, ThresholdAllowance}
+import uk.gov.hmrc.merchandiseinbaggage.stubs.PayApiStub._
+import views.html._
+import uk.gov.hmrc.scalatestaccessibilitylinter.views.AutomaticAccessibilitySpec
+
+class FrontendAccessibilitySpec
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply a11y/uk/gov/hmrc/merchandiseinbaggage/FrontendAccessibilitySpec.scala]: formatted

File: a11y/uk/gov/hmrc/merchandiseinbaggage/FrontendAccessibilitySpec.scala [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  extends AutomaticAccessibilitySpec {
+
+  private val appConfig: AppConfig                                 = app.injector.instanceOf[AppConfig]
+  private val emailForm: Form[Email] = EnterEmailForm.mandatoryForm
+  private val optionEmailForm: Form[Option[Email]] = EnterEmailForm.optionalForm
+  private val booleanForm: Form[Boolean] = Form("value" -> boolean)
+
+  implicit val arbHtml: Arbitrary[Html]                             = fixed(Html(""))
+  implicit val arbForm: Arbitrary[Form[_]] = fixed(booleanForm)
+  implicit val arbAppConfig: Arbitrary[AppConfig]                   = fixed(appConfig)
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply a11y/uk/gov/hmrc/merchandiseinbaggage/FrontendAccessibilitySpec.scala]: added implicit Form[Stirng]. Maybe we can look at refactoring strings as another story.

File: README.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 # merchandise-in-baggage-frontend
 
-**Who uses the repo/service**
+##Who uses the repo/service
```
Comment (frikit) [inline README.md:3]: Need space `## Who`

File: README.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 
 Business travellers carrying commercial goods for both import or export.
 
-**How to start the service locally**
+##How to start the service locally
```
Comment (frikit) [inline README.md:7]: Need space `## How`

File: README.md [hunk 6]
```diff
@@ -19,14 +19,14 @@ This will start all the required services to complete a journey
 `local url` http://localhost:8281/declare-commercial-goods/start-import
 `local url` http://localhost:8281/declare-commercial-goods/start-export
 
-**How to run tests**
+##How to run tests
```
Comment (frikit) [inline README.md:22]: Need space `## How`

File: README.md [hunk 7]
```diff
@@ -19,14 +19,14 @@ This will start all the required services to complete a journey
...
 
 `./run_all_tests` will run all the tests, including unit, UI and consumer contract tests. The consumer tests will generate
 contract files stored in the project root directory folder `pact`
 The generated contracts test will then being used from the Backend contract verifier by running the script:
 `checkincheck.sh`. However, currently contracts test only runs for local build.
 
-**UI Tests Repo**
+##UI Tests Repo
```
Comment (frikit) [inline README.md:29]: Need space `## UI`

File: README.md [hunk 8]
```diff
@@ -48,3 +48,13 @@ OR
...
+
+## Accessibility Tests
+
+### Prerequisites
+Have node installed on your machine
+
+### Execute tests
+To run the tests locally, simply run:
+```bash
+sbt
```
Comment (frikit) [inline README.md:60]: missing command itself 
And also missing License

**Review by frikit — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (lmugnai) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (lmugnai) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (lmugnai) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (lmugnai) [reply README.md]: done

File: README.md (hunk 8 above)
Comment (lmugnai) [reply README.md]: done

**Review by frikit — APPROVED** [review]

**Pull request merged**